lib/i2cscan/i2cscan.cpp

```diff
@@ -111,8 +112,16 @@ namespace I2CSCAN
     }
 
     bool isI2CExist(uint8_t addr) {
-        Wire.beginTransmission(addr);
-        byte error = Wire.endTransmission();
+        byte error;
+#if ESP32C3
+        do {
+#endif
+            Wire.beginTransmission(addr);
+            error = Wire.endTransmission();
+#if ESP32C3
+        }
+        while (error == 5);
```

Comment:
Do we need a timeout here if the bus is locked up like SCL tied to ground?

Reply:
You were right, it can freeze on shorted bus. Added try counter. Tested with bmi160 and it require only one try.